subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -425,12 +420,114 @@ static void prio_recv_thread(void *p1, void *p2, void *p3)
 		 * Blocking-take of the semaphore; we take it once ULL mayfly
 		 * has let it go in ll_rx_sched().
 		 */
-		k_sem_take(&sem_prio_recv, K_FOREVER);
+		k_sem_take(&sem_recv, K_FOREVER);
 		/* Now, ULL mayfly has something to give to us */
 		LOG_DBG("sem taken");
 	}
 }
 
+#else /* !CONFIG_BT_CTLR_RX_PRIO_STACK_SIZE */
+static void node_rx_recv(const struct device *dev)
+{
+#if defined(CONFIG_BT_CONN) || defined(CONFIG_BT_CTLR_ADV_ISO)
+	const struct hci_driver_data *data = dev->data;
+#endif /* CONFIG_BT_CONN || CONFIG_BT_CTLR_ADV_ISO */
```

Comment:
Not quite sure I understand your last comment here. Looks to me that this can be moved into the 
```
		while (num_cmplt != 0U) {
#if defined(CONFIG_BT_CONN) || defined(CONFIG_BT_CTLR_ADV_ISO)
```
part to reduce scope

Reply:
`dev->data->recv()`  .... this is double dereferencing, two indirect address be loaded to call a function.

Keeping one outside the loop will need single indirect address loaded to call the function (similar to a function pointer that it is already) in the loop.

Reply:
Ah, so if I understand you correct, this is done to not dereference `dev` for each loop iteration? 

I was under the impression that the compiler would not do that, especially not when the expression is `const`, but I guess that since `dev->data` can in theory be changed per iteration from another thread, it cannot do that.

Reply:
>if I understand you correct, this is done to not dereference dev for each loop iteration?

Yes.  In the `struct device` `*data` is not const. Though in this content `data` or `*data` is not being changed, the compiler does not know that, hence we make it obvious for the compiler by using `const struct hci_driver_data *data = dev->data;`